Documentation layout documentation (#2996)
canonical-web-and-design/vanilla-framework: templates/docs/layouts/documentation.md

---
wrapper_template: '_layouts/docs.html'
context:
  title: Documentation | Layouts
---

## Documentation layout

<hr>

> Documentation page layout can be build using Vanilla grid and common components. It consists of header with main navigation, optional hero strip (that may contain search field), grid based content area and a footer.

Reviewer: ```suggestion
Documentation page layout can be built using the Vanilla frameworks grid and common components. It consists of a header containing main navigation, optional hero strip (that may contain search field), grid-based content area and a footer.
```

### Structure

#### Heading

> Heading with the main navigation is build with [navigation component](/docs/patterns/navigation#global-navigation).

Reviewer: ```suggestion
The heading with the main navigation, built with the [navigation component](/docs/patterns/navigation#global-navigation).
```

> Style and contents of documentation main navigation should be consistent with rest of the site.

Reviewer: ```suggestion
Style and contents of the documentation main navigation should be consistent with rest of the site.
```

Documentation pages may have an optional search box in the main navigation.

#### Hero

> Documentation pages can have a hero area above main content area. This part of the page will usually contain a search field.

Reviewer: ```suggestion
Documentation pages can have a hero area above the main content area. This part of the page will usually contain a search field.
```

> Hero area is build with strip component with grid row inside. Usually it would be shallow light strip (`.p-strip--light is-shallow`), but the specific styling can be customised to match the site branding or other design requirements.

Reviewer: ```suggestion
The hero area is built with the [Strip component]https://vanillaframework.io/docs/patterns/strip) with grid row inside. Usually, it would be shallow light strip (`.p-strip--light is-shallow`), but the specific styling can be customised to match the site branding or other design requirements.
```

#### Content area

> Content area is placed inside a regular strip (`.p-strip`) and a grid row (`.row`). Within standard Vanilla 12 column grid, 3 of the columns are reserved for the side navigation (`.col-3`) with the rest of the row width (9 columns, `.col-9`) reserved for main documentation content.

Reviewer: ```suggestion
The content area is placed inside a regular strip (`.p-strip`) and a grid row (`.row`). Within the standard Vanilla 12 column grid, 3 of the columns are reserved for the side navigation (`.col-3`) with the rest of the row width (9 columns, `.col-9`) is deadicated to the main documentation content.
```

Sidebar should contain only the [side navigation component](/docs/patterns/navigation#side-navigation) with a list of all documentation pages. Grouping and nesting of navigation items in side navigation component should be used to build the logical structure of documentation navigation. Side navigation component has built-in responsive functionality which makes the sidebar expandable on small screens.

> Main navigation area is placed in `col-9` grid container. Within this area 9 grid columns are available. For most of documentation content standard flow of the document should be enough in the main content. Default base styling of Vanilla will provide proper spacing for the documentation content.

Reviewer: ```suggestion
The main content area is placed in `col-9` grid container. Within this area 9 grid columns are available. For most of the documentation content standard flow of the document should be enough in the main content. Default base styling of Vanilla will provide proper spacing for the documentation content.
```

---
wrapper_template: '_layouts/docs.html'
context:
  title: Documentation | Layouts
---

## Documentation layout

<hr>

### Structure

> The documentation page layout can be built using the Vanilla grid classes and common components. It consists of 3 horizontal areas that span the entire fixed width of the grid: header, content, footer.

Reviewer: ```suggestion
The documentation layout is built using Vanilla grid classes and common components. It consists of 3 horizontal areas that span the entire fixed width of the grid: header, content, footer.
```

![Documentation layout structure](https://assets.ubuntu.com/v1/cdf09557-Documentation+layout%401x.svg)

At the large breakpoint, the content area is further divided into an aside (3 columns) and a main content area (9 columns).

At smaller breakpoints, the aside is moved offscreen and shown / hidden using a toggle.

#### Header

> The header with the main navigation, built with the [navigation component](/docs/patterns/navigation#global-navigation).

Reviewer 2: How about:

"Place the main navigation and any other full width elements in the header. This could include a strip with a search, a hero element, etc."

> Style and contents of the documentation main navigation should be consistent with rest of the site.

Reviewer 2: Not sure what this is meant to highlight?

Author: Just to make sure that the main navigation is the same across the site. We had the cases (Vanilla was one of those), where main nav was inconsistent between docs and rest of the page.

##### Search

Documentation pages may have an optional search box in the main navigation.

> Alternatively search can be added in a full-width area under the top navigation, but above the aside and main content in a [strip component](/docs/patterns/strip) with grid row inside. Usually, it would be shallow light strip (`.p-strip--light is-shallow`), but the specific styling can be customised to match the site branding or other design requirements.

Reviewer 2: ```suggestion
Alternatively search can be added in a full-width area under the top navigation, but above the aside and main content in a [strip component](/docs/patterns/strip) with a grid row inside.
```

Reviewer 2: Can we be more generic about the type of strip? I just advised @danielmutis to use a regular strip around a searchbox, because the content above (slanted suru) and below (logos) required a bit more separation.

> Alternatively search can be added in a full-width area under the top navigation, but above the aside and main content in a [strip component](/docs/patterns/strip) with grid row inside. Usually, it would be shallow light strip (`.p-strip--light is-shallow`), but the specific styling can be customised to match the site branding or other design requirements.

Reviewer: ```suggestion
Alternatively, a search can be added in a full-width area under the top navigation, but above the aside and main content in a [strip component](/docs/patterns/strip) with grid row inside. Usually, it would be shallow light strip (`.p-strip--light is-shallow`), but the specific styling can be customised to match the site branding or other design requirements.
```

#### Content area

The content area is implemented as a regular strip (`.p-strip`) with a grid row (`.row`) inside. Within the standard Vanilla 12 column grid, 3 of the columns are reserved for the side navigation (`.col-3`) with the rest of the row width (9 columns, `.col-9`) is deadicated to the main documentation content.

> Sidebar should contain only the [side navigation component](/docs/patterns/navigation#side-navigation) with a list of all documentation pages. Grouping and nesting of navigation items should be used to build the logical structure of documentation navigation. Side navigation component has built-in responsive functionality which makes the sidebar expandable on small screens.

Reviewer 2: Did we make the decision to go with sidebar vs aside? I'm ok either way, just checking if we're naming it consistently.

> Sidebar should contain only the [side navigation component](/docs/patterns/navigation#side-navigation) with a list of all documentation pages. Grouping and nesting of navigation items should be used to build the logical structure of documentation navigation. Side navigation component has built-in responsive functionality which makes the sidebar expandable on small screens.

Reviewer 2: ```suggestion
The sidebar should contain the [side navigation component](/docs/patterns/navigation#side-navigation) with a list of all documentation pages. Grouping and nesting of navigation items should be used to build the logical structure of the documentation navigation. The side navigation component has built-in responsive functionality which makes it appear / go offscreen using a toggle.
```

> The main content area is placed in `col-9` grid container. Within this area 9 grid columns are available. For most of the documentation content standard flow of the document should be enough in the main content. Default base styling of Vanilla will provide proper spacing for the documentation content.

Reviewer 2: ```suggestion
The main content area is placed in a `col-9` grid container. Note that the number of columns available to use by content inside this container is equal to the number of columns the container spans. For the main content this means 9 available columns. 
```

> In cases when grid is used in documentation content for some specific layout, the maximum of 9 columns should be used.

Reviewer 2: ```suggestion
A visualisation of the grid, and how to nest different layouts inside the main content container:
```

<div class="grid-demo">
  <div class="row">
    <div class="col-1">.col-1</div>
    <div class="col-1">.col-1</div>
    <div class="col-1">.col-1</div>
    <div class="col-1">.col-1</div>
    <div class="col-1">.col-1</div>
    <div class="col-1">.col-1</div>
    <div class="col-1">.col-1</div>
    <div class="col-1">.col-1</div>
    <div class="col-1">.col-1</div>
  </div>
</div>

> To create 3 part split of the page use `col-3`:

Reviewer 2: ```suggestion
To split the main content area into 3 parts, use `col-3`:
```

<div class="grid-demo">
  <div class="row">
    <div class="col-3">.col-3</div>
    <div class="col-3">.col-3</div>
    <div class="col-3">.col-3</div>
  </div>
</div>

> To create 2 part split of the page use `col-4`:

Reviewer 2: ```suggestion
To split the main content into 2 parts, use `col-4`:
```

<div class="grid-demo">
  <div class="row">
    <div class="col-4">.col-4</div>
    <div class="col-4">.col-4</div>
  </div>
</div>

#### Footer

> Footer is build with a [strip component](/docs/patterns/strip). Footer documentation pages should be consistent with the rest of the site.

Reviewer 2: ```suggestion
The footer is built using a [strip component](/docs/patterns/strip).
```

### Example

<div class="embedded-example"><a href="/docs/examples/layouts/documentation/" class="js-example" data-height="600">

> View example of the documentation layout

Reviewer 2: ```suggestion
View an example of the documentation layout
```
